managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorContainer.java

```diff
@@ -164,6 +175,22 @@ public ManagedCursor getSlowestReader() {
         }
     }
 
+    public boolean hasCursors() {
+        long stamp = rwLock.tryOptimisticRead();
+        boolean isEmpty = cursors.isEmpty();
+        if (!rwLock.validate(stamp)) {
+            // Fallback to read lock
+            stamp = rwLock.readLock();
+            try {
+                isEmpty = cursors.isEmpty();
```

Comment:
Wouldn't the logic be the reverse here? We're checking `isEmpty` but the method should return true if it's not empty, no?

Reply:
Ya just used the same logic as "isEmpty" but the name of the method hints at a different semantic meaning.